### ISSUE TITLE
A bug that refers to the same service in multiple protocols at runtime.

### ISSUE THREAD

**Issue by NeGnail** [alipay/sofa-rpc-boot-projects#33]
For example：
```xml

    <sofa:reference id="lazyServiceReferenceBolt" interface="com.alipay.sofa.rpc.samples.lazy.LazyService">
        <sofa:binding.bolt/>
    </sofa:reference>

    <sofa:reference id="lazyServiceReferenceDubbo" interface="com.alipay.sofa.rpc.samples.lazy.LazyService">
        <sofa:binding.dubbo/>
    </sofa:reference>
```

Because the same service is referenced by the bolt and the dubbo protocol, the startup is in an infinite wait.

**Comment by NeGnail** [comment]
Because at the start, the runtime can be in registered components to `com.alipay.sofa.rpc.boot.runtime.binding.RpcBinding#getBindingHashCode` method as a unique identifier.
See the methods:
`com.alipay.sofa.runtime.service.component.ReferenceComponent#ReferenceComponent`。

Because the service reference for the different protocols is the same as the hashcode returned at this point, the subsequent service reference is registered to runtime with a wait in that method.
`com.alipay.sofa.runtime.service.component.ReferenceComponent#getImplementation`。


Now modify the `com.alipay.sofa.rpc.boot.runtime.binding.RpcBinding#getBindingHashCodemethod`, the agreement also will be considered for calculating hashcode logic.

**Comment by NeGnail** [comment]
fixed  [#27](https://github.com/alipay/sofa-rpc-boot-projects/pull/27)
